user-management/src/main/java/de/tum/in/www1/artemis/usermanagement/security/ArtemisInternalAuthenticationProvider.java

```diff
@@ -0,0 +1,91 @@
+package de.tum.in.www1.artemis.usermanagement.security;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.connectors.ConnectorHealth;
+import de.tum.in.www1.artemis.service.user.PasswordService;
+import de.tum.in.www1.artemis.usermanagement.service.user.UserCreationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.security.authentication.InternalAuthenticationServiceException;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Component
+@ConditionalOnProperty(value = "artemis.user-management.use-external", havingValue = "false")
+public class ArtemisInternalAuthenticationProvider extends ArtemisAuthenticationProviderImpl implements ArtemisAuthenticationProvider {
+
+    private final Logger log = LoggerFactory.getLogger(ArtemisInternalAuthenticationProvider.class);
+
+    public ArtemisInternalAuthenticationProvider(UserRepository userRepository, PasswordService passwordService, UserCreationService userCreationService) {
+        super(userRepository, passwordService, userCreationService);
+    }
```

Comment:
[nit] isn't the constructor redundant? It is only passing the arguments to superclass constructor call.

Reply:
If I remove it, I get an error that feault constructor is missing in the extended class